docs/embeddings/roboflow.md

```diff
@@ -0,0 +1,142 @@
+
+### Step #2: Create a Chroma Vector Database
+
+To load and save image embeddings into Chroma, we first need images to embed. In this guide, we are going to use the COCO 128 dataset, a collection of 128 images from the Microsoft COCO dataset. This dataset is available on Roboflow Universe, a community that has shared more than 250,000 public computer vision datasets.
+
+To download the dataset, visit the COCO 128 web page, click “Download Dataset” and save the dataset as a ZIP file following the on-screen instructions:
+
+![COCO 128 dataset](https://media.roboflow.com/coco128.png)
+
+We are going to use the images in the `train/images` folder of the dataset.
+
+Now that we have a dataset ready, we can create a vector database and start loading embeddings.
+
+Install the Chroma Python client with the following command:
+
+```
+pip install chromadb
+```
+
+Then, create a new Python file and add the following code:
+
+```python
+import chromadb
+import os
+from chromadb.utils.embedding_functions import RoboflowEmbeddingFunction
+import uuid
+
+client = chromadb.PersistentClient(path="database")
+
+collection = client.create_collection(name="images", metadata={"hnsw:space": "cosine"})
+
+IMAGE_DIR = "COCO128/train/images/"
+SERVER_URL = "https://infer.roboflow.com"
+API_KEY = “”
+
+ef = RoboflowEmbeddingFunction(API_KEY, api_url = SERVER_URL)
+
+documents = [os.path.join(IMAGE_DIR, img) for img in os.listdir(IMAGE_DIR)]
+embeddings = ef(images = [img for img in documents])
+ids = [str(uuid.uuid4()) for _ in range(len(documents))]
+
+print(embeddings)
+
+collection.add(
+    embeddings=embeddings,
+    documents=documents,
+    ids=ids,
+)
+```
+
+In this code snippet, we create a new Chroma database called `images`. Our database will use cosine similarity for embedding comparisons.
+
+We calculate CLIP embeddings for all images in the `COCO128/train/images` folder using Inference. We save the embeddings in Chroma using the `collection.add()` method.
+
+We store the file names associated with each image in the `documents` variable, and embeddings in `embeddings`. 
+
+If you want to use the hosted version of Roboflow Inference to calculate embeddings, replace the `SERVER_URL` value with `https://infer.roboflow.com`. We use the RoboflowEmbeddingFunction, built in to Chroma, to interact with Inference.
+
+Run the script above to calculate embeddings for a folder of images and save them in your database.
+
+We now have a vector database that contains some embeddings. Great! Let’s move on to the fun part: running a search query on our database.
+
+### Step #3: Run a Search Query
+
+To run a search query, we need a text embedding of a query. For example, if we want to find vegetables in our collection of 128 images from the COCO dataset, we need to have a text embedding for the search phrase “baseball”.
+
+To calculate a text embedding, we can use Inference through the embedding function we defined earlier:
+
+```python
+query = ef("baseball")
```

Comment:
if you have the embedding function set on the collection, it is not necessary to first embed the query.